### ISSUE TITLE
others array not working

### ISSUE THREAD

**Issue by rhs615** [Error-freeIT/Dock-Master#4]
In OSX 10.12.4 (at least thats what I'm testing with) - the others array is being ignored. No matter what combination of adding in the Applications or Downloads folder seems to work. Even deploying the dockmastertemplate.plist file will not load. Any ideas or ways I can help?

**Comment by Error-freeIT** [comment]
Hello rhs615,
I am not sure what you mean by:
> Even deploying the dockmastertemplate.plist file will not load

The dockmastertemplate.plist file is only used with the dockmaster command line tool, you do not deploy that plist file to your clients.


To quote the [README.md](https://github.com/Error-freeIT/Dock-Master):

```
1) Install the latest Dock Master.pkg.

2) Customise the dockmastertemplate.plist in Xcode or your favourite text editor. See the Dock Master Template Options table below for further details.

3) Open Terminal:

To generate a profile: dockmaster -m -t dockmastertemplate.plist /destination/profile.mobileconfig
To generate a package template: dockmaster -p -t dockmastertemplate.plist /destination/
```

If it's still not working for you please attach your customised dockmastertemplate.plist and I'll take a look at it.

Thank you
Michael

**Comment by rhs615** [comment]
Hi Michael,

Sorry - should have been more clear, but I think I have also figured out
the issue:

I was using DockMaster to get the .mobileconfig profile which I uploaded to
the JSS. I seemed to have the following issues, regardless of whether I
used the template .plist to build it or if I modified it.

The tile_size would not work - the icons were incredibly small.
No matter what I tried I could not get the others array to work  (to show
Applications or Downloads)

Turns out when uploading to Jamf Pro, the dock payload in the Configuration
Profile would overwrite portions - specifically size and others
(Applications and Downloads). Now, I can modify the
dockmastertemplate.plist, build a Configuration Profile using DockMaster,
upload to the JSS and then change the Icon Size and check the boxes for
showing Application/Downloads on the dock within the Jamf Pro UI. Now it
works like a charm!

Thanks for your help and your contribution here!

Randy

--
Randall Shore
Director of Information Technology

Office of Technology Services
Columbian College of Arts and Sciences
The George Washington University
801 22nd Street, NW, Suite 355
Washington, DC 20052

(202) 994-2132 Office
(202) 994-8096 Help Desk

http://columbian.gwu.edu/ots
